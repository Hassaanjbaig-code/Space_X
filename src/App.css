* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;

  --color-primary: #0290ff;
  --color-primary-dark: #0290ff;
  --color-text: #121212;
  --color-text-blue: #227fdc;
  --color-text-gray: #888;
}

.flex-center {
  display: flex;
  justify-content: center;
  align-items: center;
}

.row {
  display: flex;
}

.column {
  display: flex;
  flex-direction: column;
}

.no-style {
  list-style: none;
  text-decoration: none;
  background-color: transparent;
  border: none;
}

.seprator {
  height: 15px;
  border: 1px solid #121212;
}

.App {
  text-align: center;
}
