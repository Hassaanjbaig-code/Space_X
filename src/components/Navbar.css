.navbar {
  justify-content: space-between;
  padding: 12px;
  
}

.navbar img {
  width: 40px;
  height: 40px;
  margin-right: 12px;
}

.list-nav {
  gap: 8px;
  font-size: 0.8rem;
  color: var(--color-text-blue);
}

.seprator {
  height: 15px;
  border: 1px solid #121212;
}
