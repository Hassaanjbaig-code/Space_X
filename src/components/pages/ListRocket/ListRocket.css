.rocket {
  padding: 24px;
  width: 100%;
  text-align: start;
}

.rocket img {
  width: 250px;
  height: 150px;
}

.rocket-description {
  padding-left: 12px;
}

.rocket button {
  width: fit-content;
  background-color: blue;
  padding: 8px;
  color: white;
  border: none;
  border-radius: 4px;
}
