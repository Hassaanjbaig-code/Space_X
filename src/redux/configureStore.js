import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './rockets/rocketReducer';
import missionReducer from './missions/missionReducer';

export default configureStore({
  reducer: {
    rocketReducer,
    mission: missionReducer.reducer,
  },
});
