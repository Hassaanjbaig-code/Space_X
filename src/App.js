import './App.css';
import Navbar from './components/Navbar';
import { Routes, Route } from 'react-router-dom';
import ListMission from './components/pages/ListMission';
import ListRockect from './components/pages/ListRockect';
import Profile from './components/pages/Profile';

function App() {
  return (
    <div className="App">
      <Navbar />
      <Routes>
        <Route path="/" element={<ListRockect />} />
        <Route path="/Missions" element={<ListMission />} />
        <Route path="/My Profile" element={<Profile />} />
      </Routes>
    </div>
  );
}

export default App;