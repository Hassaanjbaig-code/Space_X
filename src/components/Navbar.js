import React from 'react';
import logo from '../assets/images/planet.png';
import './Navbar.css';

function Navbar() {
  return (
  <nav className='flex-center navbar'>
    <div className="row logo">
      <img src={logo} alt="Logo Space X" />
      <h1>Space Travelers'Hub</h1>
    </div>
    <ul className='flex-center no-style list-nav'>
      <li>Rockets</li>
      <li>Missions</li>
      <hr className='seprator'/>
      <li>My Profile</li>
    </ul>
  </nav>
  )
}

export default Navbar