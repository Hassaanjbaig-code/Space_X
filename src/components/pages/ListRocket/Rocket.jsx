import React from 'react'
import { useDispatch } from 'react-redux'
import { reserveRocket, unreserveRocket } from '../../../redux/rockets/rocketReducer'
import './ListRocket.css'
import PropTypes from 'prop-types';

const Rocket = (props) => {
  const dispatch = useDispatch()
  const {id, flickr_images, rocket_name, 
    description, reserved} = props.rocket

  
  return (
    <div className='row rocket'>
      <img src={flickr_images} alt={"Rocket " + rocket_name } />
      <div className="column rocket-description">
        <h4>{rocket_name}</h4>
        <p>{reserved && (<span className='badge-reserved'>Reserved</span>)} {description}</p>
        {reserved? 
         <button onClick={()=>{dispatch(unreserveRocket(id))}}>Cancel reservation</button>
        : <button onClick={()=>{dispatch(reserveRocket(id))}}>Reserve Rocket</button>}
      </div>
    </div>
  )
}

export default Rocket