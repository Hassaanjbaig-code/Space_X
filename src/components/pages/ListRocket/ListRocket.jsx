import React from 'react';
import Rocket from './Rocket';
import './ListRocket.css'
import { useEffect } from 'react';
import { shallowEqual, useDispatch, useSelector } from 'react-redux'
import { fetchRockets } from '../../../redux/rockets/rocketReducer';


const ListRocket = () => {
  const dispatch = useDispatch()

  useEffect(() => {
    dispatch(fetchRockets())
  }, [])

  const listRocket = useSelector(state => state.rocketReducer.listRocket, shallowEqual)

  return (
    <ul>
      {(listRocket)? listRocket.map(rocket => (
        <li key={rocket.id} ><Rocket rocket={rocket}/></li>
      )): <p>Empty list</p>}
    </ul>
  )

};

export default ListRocket;
