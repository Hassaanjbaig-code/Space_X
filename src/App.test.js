
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom'
import { Provider } from 'react-redux';
import App from './App';
import configureStore from './redux/configureStore';
import { BrowserRouter as Router } from 'react-router-dom';
import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import ListRocket from './components/pages/ListRocket/ListRocket';
import Navbar from './components/Navbar';

const store = configureStore
describe('test render', () => {
  test('List Rocket should match snapshoot', () => { 
    const tree = renderer.create(
      <React.StrictMode>
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>
    </React.StrictMode>,
    ).toJSON()
    
    expect(tree).toMatchSnapshot();
  })
})
