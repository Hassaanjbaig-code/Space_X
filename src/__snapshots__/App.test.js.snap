// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render List Rocket should match snapshoot 1`] = `
<div
  className="App"
>
  <nav
    className="flex-center navbar"
  >
    <div
      className="row logo"
    >
      <img
        alt="Logo Space X"
        src="planet.png"
      />
      <h1>
        Space Travelers Hub
      </h1>
    </div>
    <ul
      className="flex-center no-style list-nav"
    >
      <li>
        <a
          aria-current="page"
          className="navLink active"
          href="/"
          onClick={[Function]}
          path="/"
        >
          Rockets
        </a>
      </li>
      <li>
        <a
          className="navLink"
          href="/missions"
          onClick={[Function]}
          path="/missions"
        >
          Missions
        </a>
      </li>
      <hr
        className="seprator"
      />
      <li>
        <a
          className="navLink"
          href="/myprofile"
          onClick={[Function]}
          path="/myprofile"
        >
          My Profile
        </a>
      </li>
    </ul>
  </nav>
  <ul>
    <p>
      Empty list
    </p>
  </ul>
</div>
`;
